columnphysics/icepack_therm_itd.F90

```diff
@@ -2140,6 +2138,9 @@ subroutine icepack_step_therm2 (dt, ncat, nltrcr,           &
       !-----------------------------------------------------------------
 
        if (icepack_chkoptargflag(first_call)) then
+          if (present(update_ocn_f)) then
+             call icepack_init_parameters(update_ocn_f_in=update_ocn_f)
+          endif
```

Comment:
This new code needs to be outside icepack_chkoptargflag.  That interface is for checking whether optional arguments that are needed are passed, and the settings reflect never checking to always checking.  The icepack_init_parameters call should be outside that if block.  If a user passes update_ocn_f in, then the code should update the value in parameters.  Otherwise, just continue.  And it might vary by gridcell or timestep (even if it shouldn't).